drivers/net/wireless/marvell/mwifiex/pcie_quirks.h

```diff
@@ -3,102 +3,17 @@
  * Header file for PCIe quirks.
  */
 
-#include <linux/dmi.h>
+#include "pcie.h"
 
-/* List of Surface gen4+ (Pro 4 or later gen) devices that have mwifiex */
-static const struct dmi_system_id surface_gen4p_dmi_table[] = {
-	{
-		.ident = "Surface Pro 4",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Surface Pro 4"),
-		},
-	},
-	{
-		.ident = "Surface Pro 5",
-		.matches = {
-			/* match for SKU here due to generic product name "Surface Pro" */
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_SKU, "Surface_Pro_1796"),
-		},
-	},
-	{
-		.ident = "Surface Pro 5 (LTE)",
-		.matches = {
-			/* match for SKU here due to generic product name "Surface Pro" */
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_SKU, "Surface_Pro_1807"),
-		},
-	},
-	{
-		.ident = "Surface Pro 6",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Surface Pro 6"),
-		},
-	},
-	{
-		.ident = "Surface Book 1",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Surface Book"),
-		},
-	},
-	{
-		.ident = "Surface Book 2",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Surface Book 2"),
-		},
-	},
-	{
-		.ident = "Surface Laptop 1",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Surface Laptop"),
-		},
-	},
-	{
-		.ident = "Surface Laptop 2",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Surface Laptop 2"),
-		},
-	},
-	{}
-};
-
-/* Surface 3 */
-static const struct dmi_system_id surface3_dmi_table[] = {
-	{
-		.ident = "Surface 3",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Surface 3"),
-		},
-	},
-	{
-		.ident = "Surface 3",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_BIOS_VENDOR, "American Megatrends Inc."),
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "OEMB"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "OEMB"),
-		},
-	},
-	{}
-};
-
-/* Surface Pro 3 */
-static const struct dmi_system_id surfacepro3_dmi_table[] = {
-	{
-		.ident = "Surface Pro 3",
-		.matches = {
-			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
-			DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "Surface Pro 3"),
-		},
-	},
-	{}
-};
+/* Quirks */
+#define QUIRK_NONE		0
```

Comment:
I think you can drop this. If there isn't any bit set it should be implied that there's no quirk required.

Reply:
I thought I can use `QUIRK_NONE` when adding quirk implementation like `.driver_data = (void *)QUIRK_NONE,` (and the actual quirks will come in later commits).

That said, I can't find the real usage in upstream tree yet. In this case, using `0` directly is preferred? (`.driver_data = 0,`). I'll keep the `QUIRK_NONE` for now.

Reply:
I think `= 0` directly is fine here.

Reply:
I think `= 0` is fine here.

Although that's really only my own preference. I think that bit-flags imply that zero is none, the same way you don't do `FLAG_NONE | FLAG_THING_ONE | FLAG_THING_TWO` and just `FLAG_THING_ONE | FLAG_THING_TWO`.

Also since it's static data, you could also drop the `.driver_data = 0` entirely as the default initialization covers that and will initialize it to zero.

Reply:
> I think `= 0` is fine here.
> 
> Although that's really only my own preference. I think that bit-flags imply that zero is none, the same way you don't do `FLAG_NONE | FLAG_THING_ONE | FLAG_THING_TWO` and just `FLAG_THING_ONE | FLAG_THING_TWO`.

OK.

> Also since it's static data, you could also drop the `.driver_data = 0` entirely as the default initialization covers that and will initialize it to zero.

The first patch is a implementation, I want to preserve the skeleton as much as possible. Also I see a few usage of `.driver_data = 0` in upstream. So, I'll go this way (use `.driver_data = 0` and drop `QUIRK_NONE`).

Reply:
Addressed with the new rebase.